### PR TITLE
Add private members to argparse.pyi.

### PR DESCRIPTION
**Pull request by rchen152**
These attributes are sometimes used despite being private, especially the Action subclasses.

Based on https://github.com/python/cpython/blob/master/Lib/argparse.py.

### CODE REVIEW

**Comment by matthiaskramm** [comment]
Is there some issue with mypy travis right now? We're seeing:

```
FAILURE  #5 check package mypy
mypy/main.py:235: note: unused 'type: ignore' comment
mypy/main.py:240: note: unused 'type: ignore' comment
```

**Comment by JelleZijlstra** [comment]
Those lines are like `        arg = parser.add_argument(flag,  # type: ignore  # incorrect stub for add_argument`. It looks like this PR fixes the incorrect stubs.

We should merge this (after it's reviewed), then immediately make a PR to mypy syncing typeshed and removing the type: ignores.

**Comment by rchen152** [comment]
*ping*
Could I get a review of this soon?

File: stdlib/2and3/argparse.pyi [hunk 1]
```diff
@@ -1,41 +1,82 @@
...
 from typing import (
-    Any, Callable, Iterable, List, IO, Optional, Sequence, Tuple, Type, Union,
-    TypeVar, overload
+    Any, Callable, Dict, Generator, Iterable, List, IO, NoReturn, Optional,
+    Pattern, Sequence, Tuple, Type, Union, TypeVar, overload
 )
 import sys
 
 _T = TypeVar('_T')
+_ActionVar = TypeVar('_ActionVar', bound='Action')
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi]: Usually we suffix typevars with `T` (e.g., `_ActionT`).

File: stdlib/2and3/argparse.pyi [hunk 2]
```diff
@@ -1,41 +1,82 @@
...
+    _negative_number_matcher: Pattern[str]
+    _has_negative_number_optionals: List[bool]
+
+    def __init__(self, description: Optional[_Text], prefix_chars: _Text,
+                 argument_default: Optional[_Text], conflict_handler: _Text) -> None: ...
+    def register(self, registry_name: _Text, value: Any, object: Any) -> None: ...
+    def _registry_get(self, registry_name: _Text, value: Any, default: Any = ...) -> Any: ...
+    def set_defaults(self, **kwargs: Any) -> None: ...
+    def get_default(self, dest: _Text) -> Any: ...
+    def add_argument(self, *args: Any, **kwargs: Any) -> Action: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi]: Why did you remove the more specific arguments to this function in the previous stub?

File: stdlib/2and3/argparse.pyi [hunk 3]
```diff
@@ -140,13 +221,17 @@ class Action:
...
+class Namespace(_AttributeHolder):
     def __init__(self, **kwargs: Any) -> None: ...
     def __getattr__(self, name: _Text) -> Any: ...
     def __setattr__(self, name: _Text, value: Any) -> None: ...
     def __contains__(self, key: str) -> bool: ...
 
 class FileType:
+    _mode: _Text
+    _bufsize: int
+    _encoding: Optional[_Text]
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi]: Do these exist in Python 2? The construction signatures below suggests they don't.

File: stdlib/2and3/argparse.pyi (hunk 2 above)
Comment (rchen152) [reply stdlib/2and3/argparse.pyi]: Those were incorrect.  I'm actually not entirely sure where they came from, given that in `argparse.py`, `add_argument` is actually defined using *args and **kwargs.

**Comment by rchen152** [comment]
Addressed all the comments.

File: stdlib/2and3/argparse.pyi (hunk 2 above)
Comment (JelleZijlstra) [reply stdlib/2and3/argparse.pyi]: The Python-visible signature is *args, **kwargs, but obviously in practice this method doesn't take any argument. Since it's probably the most frequently called part of argparse's interface, I think it's important that we give more specific types where it's at all possible. The documentation (https://docs.python.org/3.7/library/argparse.html#argparse.ArgumentParser.add_argument) also gives more specific arguments.
